### PR TITLE
Fix no_std

### PR DESCRIPTION
**Pull request by justinmoon**
Use `core::convert` instead of `std::convert`

### CODE REVIEW

**Comment by apoelstra** [comment]
Why isn't CI catching this?

I will PR to switch to github actions, and redo our scripts and make them work. Unfortunately the build is currently failing in a few ways so it may be a bit of work.

**Comment by elichai** [comment]
> Why isn't CI catching this?

Because we only use this macro in tests, where `std` is enabled anyway.
we should use this macro to create a private type that we don't use or export, just so it will test this outside of tests
(ie. `mod private {hash_newtype!(__testType, ::sha256d::Hash, 32, doc="test")}`)

**Comment by elichai** [comment]
small note, alloc only and no-std are distinctly different

**Comment by justinmoon** [comment]
Converting this to draft for now. I'm starting to add other stuff to get rust-bitcoin working on real MCUs.

**Comment by apoelstra** [comment]
can you rebase on current master? In #109 we did a fair bit of code cleanup including some nostd fixes.

**Comment by RCasatta** [comment]
I see bare-io is now called core2 and from the badge, it looks it requires 1.47+

File: src/lib.rs [hunk 1]
```diff
@@ -45,7 +48,7 @@
 
 #[macro_use] mod util;
 #[macro_use] pub mod serde_macros;
-#[cfg(any(test, feature = "std"))] mod std_impls;
+pub mod impls;
```
Comment (dr-orlovsky) [inline src/lib.rs:51]: Why we need this to be `pub`? It does not define any types to export

**Review by dr-orlovsky — COMMENTED** [review]
What is the current status on this taking into account all other work on `no_std` with bringing `core2` support?

**Comment by apoelstra** [comment]
cc @justinmoon I think we've decided to go with core2 rather than bare-io, see recently-merged #107 and #130.

Should we close this PR or would you like us to revisit it?

**Comment by TheBlueMatt** [comment]
bare-io was renamed core2. core2 is just the new version (though still beta, AFAIU). I think this can be closed.

**Comment by apoelstra** [comment]
Closing. Feel free to reopen @justinmoon if you'd like us to revisit this.

**Pull request closed without merging**